xicam/plugins/__init__.py

```diff
@@ -204,66 +230,69 @@ def _check_shadows(group, group_all):
                         f"Loading entrypoint from {winner.module_name} and ignoring others."
                     )
 
-    @threads.method(
-        threadkey="entrypoint-loader", showBusy=False, cancelIfRunning=False
-    )  # progress state managed independently
     def _load_plugins(self):
-        started_instantiating = False
-
         # For every entrypoint in the load queue
         while not self._load_queue.empty():
-            type_name, entrypoint = self._load_queue.get()
+            load_task = self._load_queue.get()
 
             # load it
-            self._load_plugin(type_name, entrypoint)
+            self._load_plugin(load_task)
 
-            if not started_instantiating:  # If this is the first load
+            if not self.instantiating:  # If this is the first load
                 # Start an event chain to pull from the queue
                 threads.invoke_as_event(self._instantiate_plugin)
-                started_instantiating = True
+                self.instantiating = True
 
             # mark it as completed
             self._load_queue.task_done()
+            yield
 
-        # Finished loading, progress
-        if self.state == State.LOADING:
-            self.state = State.INSTANTIATING
+    def _load_plugin(self, load_task:PluginTask):
+        entrypoint = load_task.entry_point
 
-    def _load_plugin(self, type_name, entrypoint: entrypoints.EntryPoint):
         # if the entrypoint was already loaded into cache and queued, do nothing
-        if self._load_cache[type_name].get(entrypoint.name, None):
+        if load_task.plugin_class is not None:
```

Comment:
I like the encapsulation of the tasks into an object so you can check if its `plugin_class` has been set to a not None value yet (instead of using the type_mapping and entrypoints to discover if the plugin has already been loaded)